charmhelpers/core/hookenv.py

```diff
@@ -468,13 +468,16 @@ def config(scope=None):
 
 
 @cached
-def relation_get(attribute=None, unit=None, rid=None):
+def relation_get(attribute=None, unit=None, rid=None, app=None):
     """Get relation information"""
     _args = ['relation-get', '--format=json']
+    if app is True:
+        _args.append('--app')
     if rid:
         _args.append('-r')
         _args.append(rid)
     _args.append(attribute or '-')
+    # unit or application name
     if unit:
         _args.append(unit)
     try:
```

Comment:
I know this convention matches the underlying `relation-get` call, but it feels really awkward.  I'm fine w/ this as-is, but I'd also like to put forth this suggested alternative UX:

```suggestion
def relation_get(attribute=None, unit=None, rid=None, app=None):
    """Get relation information"""
    _args = ['relation-get', '--format=json']
    if app is not None:
        _args.append('--app')
    if rid:
        _args.append('-r')
        _args.append(rid)
    _args.append(attribute or '-')
    # unit or application name
    if unit or app:
        _args.append(unit or app)
    try:
```

Thus, you could call it like it this:

```python
relation_get(rid=rid, app='foo')
relation_get(rid=rid, unit='foo/0')
```